packages/toast/src/ControlledToast/ControlledToast.spec.tsx

```diff
@@ -0,0 +1,171 @@
+  try {
+    await waitForElementToBeRemoved(element);
+    throw new Error('Expected to catch error.');
+  } catch (error) {
+    // eslint-disable-next-line jest/no-try-expect
+    if (error instanceof Error) {
+      expect(error.toString()).toMatch(
+        'Timed out in waitForElementToBeRemoved.',
+      );
+    }
+  }
+}
+/**
+ * This suite checks rendering specific to the controlled behavior.
+ * i.e. `open` and `onClose` props.
+ *
+ * Other individual toast behavior is tested in `InternalToast.spec`
+ * and multi-toast behavior is tested in `ToastContainer.spec`
+ */
+describe('packages/toast/controlled', () => {
+  // Clear the portals after each test
+  afterEach(() => {
+    cleanup();
+    const portal = document.querySelector('.' + toastPortalClassName);
+
+    if (portal) {
+      document.body.removeChild(portal);
+    }
+  });
+
+  describe('a11y', () => {
+    test('does not have basic accessibility issues', async () => {
+      const { container } = render(
+        <Toast open={true} onClose={() => {}} title={'hello world'} />,
+      );
+      const results = await axe(container);
+      expect(results).toHaveNoViolations();
+    });
+  });
+
+  describe('`open` prop', () => {
+    test('renders when `open` is true', async () => {
+      const { findByTestId } = render(
+        <ToastProvider>
+          <Toast
+            open={true}
+            title="Test 1"
+            onClose={() => {}}
+            data-testid="test-toast1"
+          />
+        </ToastProvider>,
+      );
+
+      const toast = await findByTestId('test-toast1');
+      expect(toast).toBeInTheDocument();
+    });
+
+    test('does not render when `open` is false', async () => {
+      const { queryByTestId } = render(
+        <ToastProvider>
+          <Toast
+            open={false}
+            title="Test 2"
+            onClose={() => {}}
+            data-testid="test-toast2"
+          />
+        </ToastProvider>,
+      );
+
+      const toast = await waitFor(() => queryByTestId('test-toast2'));
+      expect(toast).not.toBeInTheDocument();
+    });
+
+    test('is closed when `open` is set back to false', async () => {
+      const { findByTestId, rerender } = render(
+        <ToastProvider>
+          <Toast
+            open={true}
+            title="Test 3"
+            onClose={() => {}}
+            data-testid="test-toast3"
+          />
+        </ToastProvider>,
+      );
+      const toast = await findByTestId('test-toast3');
+
+      rerender(
+        <ToastProvider>
+          <Toast
+            open={false}
+            title="Test 3"
+            onClose={() => {}}
+            data-testid="test-toast3"
+          />
+        </ToastProvider>,
+      );
+
+      await waitForElementToBeRemoved(toast);
+      expect(toast).not.toBeInTheDocument();
+    });
+
+    // eslint-disable-next-line jest/expect-expect
+    test('does not close until manually closed', async () => {
```

Comment:
If you set a `timeout`, shouldn't it auto-close?

Reply:
```

```